src/main/java/com/jos/dem/vetlog/config/WebSecurityConfig.java

```diff
@@ -37,20 +37,20 @@ public class WebSecurityConfig {
     @Bean
     public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
         http.authorizeHttpRequests((requests) -> requests.requestMatchers(
-                                "/",
-                                "/actuator/**",
-                                "/assets/**",
-                                "/images/**",
-                                "/home/**",
-                                "/user/**",
-                                "/recovery/**",
-                                "/breed/list",
-                                "/pet/listForAdoption",
-                                "/service/list")
-                        .permitAll()
-                        .anyRequest()
-                        .authenticated())
-                .formLogin((form) -> form.loginPage("/login").permitAll())
+                "/",
+                "/actuator/**",
+                "/assets/**",
+                "/images/**",
+                "/home/**",
+                "/user/**",
+                "/recovery/**",
+                "/breed/list",
+                "/pet/listForAdoption",
+                "/service/list")
+                .permitAll()
+                .anyRequest()
+                .authenticated())
+                .formLogin((form) -> form.loginPage("/login").failureUrl("/login?error=true").permitAll())
                 .logout((logout) -> logout.permitAll());
```

Comment:
I like this approach! It is new for me this `failureUrl`; however, I suggest running test cases in your local so that you can format the code before push.